metals/src/main/scala/scala/meta/internal/metals/Doctor.scala

```diff
@@ -326,16 +326,107 @@ final class Doctor(
+                .element("th")(_.text("Type"))
                 .element("th")(_.text("Diagnostics"))
-                .element("th")(_.text("Goto definition"))
-                .element("th")(_.text("Completions"))
-                .element("th")(_.text("Find references"))
+                .element("th")(_.text("Interactive"))
+                .element("th")(_.text("Semanticdb"))
+                .element("th")(_.text("Debugging"))
+                .element("th")(_.text("Java support"))
                 .element("th")(_.text("Recommendation"))
             )
           ).element("tbody")(html => buildTargetRows(html, targetIds))
         )
     }
+
+    // Additional explanations
+    html.element("div")(
+      _.element("p")(
+        _.text(
+          "Diagnostics:"
+        )
+      ).element("ul")(
+        _.element("li")(
+          _.text(
+            s"${Icons.unicode.check} - connected build server showing diagnostics"
+          )
+        )
+          .element("li")(
+            _.text(s"${Icons.unicode.alert} - only syntactic errors")
+          )
+      )
+    )
+
+    html.element("div")(
+      _.element("p")(
+        _.text(
+          "Interactive features (completions, hover):"
+        )
+      ).element("ul")(
+        _.element("li")(
+          _.text(
+            s"${Icons.unicode.check} - supported Scala version"
+          )
+        )
+          .element("li")(
+            _.text(s"${Icons.unicode.error} - unsupported Scala version")
+          )
+      )
+    )
+
+    html.element("div")(
+      _.element("p")(
+        _.text(
+          "Semanticdb features (references, renames, go to implementation):"
+        )
+      ).element("ul")(
+        _.element("li")(
+          _.text(
+            s"${Icons.unicode.check} - build tool creates needed semanticdb files automatically"
+          )
+        )
+          .element("li")(
+            _.text(
+              s"${Icons.unicode.error} - no needed semanticdb indexes are produced"
+            )
+          )
+      )
+    )
+
+    html.element("div")(
+      _.element("p")(
+        _.text(
+          "Debugging (run/test, breakpoints, evaluation):"
+        )
+      ).element("ul")(
+        _.element("li")(
+          _.text(
+            s"${Icons.unicode.check} - users can run or test their code with debugging capabilities"
+          )
+        )
+          .element("li")(
+            _.text(
+              s"${Icons.unicode.error} - the tool does not support debugging in this target"
+            )
+          )
+      )
+    )
+
+    html.element("div")(
+      _.element("p")(
+        _.text(
+          "Java Support:"
+        )
+      ).element("ul")(
+        _.element("li")(
+          _.text(
+            s"${Icons.unicode.check} - working non-interactive features (references, rename etc.)"
+          )
+        )
+          .element("li")(
+            _.text(s"${Icons.unicode.error} - missing semanticdb plugin")
```

Comment:
Same as the above suggestions, I think it might be better to not display these unless they are up in the table. Another thought, we should try to make them as actionable as possible. I know for example when using `scalal-cli` you get the warning that there is no semanticdb for java targets, but it sort of leaves the user with the question "well what do I need to do to get it? How do I fix this?" In this situation will there also be a recommendation attatched?

Reply:
The recommendation should show up on the right, I can add some better info there.

Reply:
Added a bit more info that it might be automatically added by the build server. I will work on sbt to add this automatically.

Reply:
It seems that we can't add it yet to sbt BSP: https://github.com/sourcegraph/lsif-java/issues/390
I added the explanation that the only supported tool is Bloop